src/hooks/useCanvasManager.ts

```diff
@@ -0,0 +1,42 @@
+import { useCallback, useEffect, useLayoutEffect, useRef } from 'react'
+import { DrawOptions, useCanvasDraw } from './useCanvasDraw';
+import { useCanvasEdit } from './useCanvasEdit';
+
+export const useCanvasManager = (opts: DrawOptions) => {
```

Comment:
Write it as `options` or `config` if you want it shorter, it improves readability